superset/assets/src/SqlLab/components/ResultSet.jsx

```diff
@@ -163,7 +171,7 @@ export default class ResultSet extends React.PureComponent {
             </Button>
           </Alert>
         </div>);
-    } else if (query.state === 'success') {
+    } else if (['rendering', 'success'].indexOf(query.state) >= 0) {
```

Comment:
And line 159, it should apply to `success` instead of `rendering`?

Reply:
yes. this part is called twice because of `componentDidUpdate()` in `ResultSet.jsx`. It is because if the state is `rendering`. I am not sure how we can change the `state` to `success` without `re-rendering`. What's your suggestion? This sound like falling into the same discussion in your comment below.